cosmos/airflow/graph.py

```diff
@@ -171,10 +171,14 @@ def build_airflow_graph(
             node=node,
             execution_mode=execution_mode,
             args=task_args,
-            use_name_as_task_id_prefix=test_behavior != TestBehavior.AFTER_EACH,
+            use_name_as_task_id_prefix=not (test_behavior == TestBehavior.AFTER_EACH and node.has_test),
         )
         if task_meta and node.resource_type != DbtResourceType.TEST:
-            if node.resource_type == DbtResourceType.MODEL and test_behavior == TestBehavior.AFTER_EACH:
+            if (
+                node.resource_type == DbtResourceType.MODEL
+                and node.has_test is True
```

Comment:
```suggestion
                and node.has_test
```

Reply:
I'll commit the suggestion with all revision